src/gax.ts

```diff
@@ -292,23 +342,148 @@ export class CallSettings {
   }
 }
 
+/**
+ * Validates passed retry options in preparation for eventual parameter deprecation
+ * converts retryRequestOptions to retryOptions
+ * then sets retryRequestOptions to null
+ *
+ * @param {CallOptions} options - a list of passed retry option
+ * @return {CallOptions} A new CallOptions object.
+ *
+ */
+
+export function checkRetryOptions(
+  options?: CallOptions,
+  gaxStreamingRetries?: boolean
+): CallOptions | undefined {
+  // options will be undefined if no CallOptions object is passed at call time
+  if (options) {
+    // if a user provided retry AND retryRequestOptions at call time, throw an error
+    if (gaxStreamingRetries) {
+      if (
+        options.retry !== undefined &&
+        options.retryRequestOptions !== undefined
+      ) {
+        throw new Error('Only one of retry or retryRequestOptions may be set');
+      } else {
```

Comment:
No `else` needed after `throw`.

Reply:
regarding this and the comment re tott, I rearranged this a bit. It's always going to have a lot of checks, but i tried to make it all have more guard checks where possible